driver/conv_driver.hpp

```diff
@@ -1275,6 +1285,27 @@ int ConvDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
             (is_int8 ? out_dev->ToGPU(q, out_int8.data()) : out_dev->ToGPU(q, out.data.data()));
     }
 
+    if(miopen::IsEnabled(MIOPEN_DRIVER_USE_GPU_REFERENCE{}))
+    {
+        if(is_fwd)
+        {
+            out_ref_dev = std::unique_ptr<GPUMem>(
+                new GPUMem(ctx, out_sz, is_int8 ? sizeof(float) : sizeof(Tref)));
+            status |= (is_int8 ? out_ref_dev->ToGPU(q, out_int8.data())
```

Comment:
Don't allocate additional memory on GPU, just re-use existing buffers. Otherwise the new method will have limited applicability.

Reply:
While re-use `in_dev` and `wei_dev` as the input tensor for the original gpu convolution, but still need another gpu buffer (here is `out_ref_dev` for fwd) for the reference kernel, as an output. So each direction we still need at least one gpu buffer as an output, hence we can compare the final result.

Reply:
We have output device buffers for each convolution under test. And we have host buffers that currently hold reference data.

After convolution executed, you can copy it's output from device buffer to the host buffer. Then you can run reference kernel and obtain reference output in the device buffer. Then, compare device buffer and host buffer as usual.

Reply:
Oh, indeed requested from @daniellowell that we will ultimately use another gpu kernel to calculate the nrms value of the GPU and reference kernel (to avoid cpu computation & memcpy gpu->cpu). hence indeed we need 2 buffers, so, I guess I still need these buffers

Reply:
@daniellowell @carlushuang I do not see substantial value of using GPU for calculating RMS. Current implementation is not a bottleneck. The cost is smaller coverage of GPU-based verification due to earlier exhaustion of device memory. **_I vote for computing RMS on CPU._**

If you instist, we can do the the following:
- Pass two buffers that contain input tensors to the verification kernel (as usual)
- Instead of allocating a buffer for output (for reference tensor), pass buffer that is already allocated and contains output of the convolution under test.
- The verification kernel computes reference tensor, but not writing it into device memory. It should use computed values for RMS calculation. The output of the kernel is scalar RMS value.

The drawbacks of the alternative approach:
- The verification kernel is not a convolution kernel anymore (so it can't be reused somewhere else).
- There is no way for transferring reference tensor to the host (for caching purposes).

Reply:
@atamazov @daniellowell A side note here. If we do not want to introduce extra gpu memory, while use gpu reference kernel as validation backend, it means it will use the same gpu data type (currently fp16/bf16/fp32). This is different from cpu, which always use fp32 to do calculation. Currently I think there should be no big difference (indeed we always cast data back to fp32 inside kernel).